hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -88,10 +88,44 @@ def type_sorting_key(t):
 
 
 def try_issubclass(thing, superclass):
-    thing = getattr(thing, "__origin__", None) or thing
-    superclass = getattr(superclass, "__origin__", None) or superclass
     try:
-        return issubclass(thing, superclass)
+        # Else we're looking at two distinct classes - which might be generics.
```

Comment:
(Minor: as a non-native it took me a while to understand what "Else" means in this context. Perhaps you want to simplify this comment?)